apps/meteor/client/hooks/useJoinRoom.ts

```diff
@@ -2,13 +2,17 @@ import type { IRoom } from '@rocket.chat/core-typings';
 import { useMutation, useQueryClient } from '@tanstack/react-query';
 
 import { sdk } from '../../app/utils/client/lib/SDKClient';
+import type { AppPreventedError } from '../lib/errors/AppPreventedError';
+import { dispatchToastMessage } from '../lib/toast';
 
 type UseJoinRoomMutationFunctionProps = {
 	rid: IRoom['_id'];
 	reference: string;
 	type: IRoom['t'];
 };
 
+const isAppError = (error: AppPreventedError | Error): error is AppPreventedError => (error as AppPreventedError).reason !== undefined;
```

Comment:
Can we assume that by only not having reason is an app error? 🤔

Reply:
In this case, yes... the app must have a reason in the response...